### DISCUSSION TITLE
Saving parsed json to (and reading it back from) memory, e.g. in binary format

### DISCUSSION THREAD

**Discussion by Tom-Evers** [nlohmann/json#3315] [Ideas]
Take this very simple example:

```cpp
json structure;
try
{
    std::ifstream i("layout.json");
    i >> structure;
}
catch (json::parse_error& e) { std::cerr << "Error parsing structure file\n" << e.what() << std::endl; }
catch (json::type_error& e)  { std::cerr << "Type error in structure file\n"  << e.what() << std::endl; }
catch (json::other_error& e) { std::cerr << "Other error in structure file\n" << e.what() << std::endl; }
```

The line `i >> structure;` parses the contents of `layout.json` into a `json` object called `structure`.

Is there any way for me to dump `structure` into some file that can be read directly back into a `json` object (i.e. without having to parse it)?

<br>
<br>

Ideally, I'd like a compile time solution. Something I can "include" directly. For example: I run the parser once and export the resulting `json` object to something like a binary blob that I would just have to `reinterpret_cast` to use.

**Comment by nlohmann** [comment]
No, there is no way to read a JSON file without parsing it. But you can similarly pipe to a `std::ofstream`, see https://json.nlohmann.me/api/basic_json/operator_ltlt/.

**Reply by Tom-Evers** [reply]
I don't want to read a JSON file without parsing it.

I want to read a JSON file, parse it, then save the _resulting data_ (i.e. the _parsed_ json) somewhere.

**Reply by nlohmann** [reply]
I understand, but this is not possible.

**Reply by Tom-Evers** [reply]
I'll let you know when I've built something for it... As it stands now, parsing a json file that contains my GUI structure takes up 40% of the startup time :(

**Reply by nlohmann** [reply]
Release mode?

**Reply by Tom-Evers** [reply]
No, debug mode, but the slow startup is kind of a big problem during development right now

**Reply by nlohmann** [reply]
Oh. How large is the JSON then? Is it deeply nested or contains of really long arrays/objects?
